src/VisualStudio/VisualBasic/Impl/CodeModel/VisualBasicCodeModelService.vb

```diff
@@ -2027,6 +2032,27 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.CodeModel
             Return False
         End Function
 
+        Public Overrides Function UpdateParameterKind(parameterKind As EnvDTE80.vsCMParameterKind, passingMode As PARAMETER_PASSING_MODE) As EnvDTE80.vsCMParameterKind
+            Dim updatedParameterKind = parameterKind
+
+            Select Case passingMode
+                Case PARAMETER_PASSING_MODE.cmParameterTypeIn
+                    updatedParameterKind = updatedParameterKind Or EnvDTE80.vsCMParameterKind.vsCMParameterKindIn
+                    updatedParameterKind = updatedParameterKind And Not EnvDTE80.vsCMParameterKind.vsCMParameterKindRef
+                    updatedParameterKind = updatedParameterKind And Not EnvDTE80.vsCMParameterKind.vsCMParameterKindOut
+
+                Case PARAMETER_PASSING_MODE.cmParameterTypeInOut
+                    updatedParameterKind = updatedParameterKind And Not EnvDTE80.vsCMParameterKind.vsCMParameterKindIn
+                    updatedParameterKind = updatedParameterKind Or EnvDTE80.vsCMParameterKind.vsCMParameterKindRef
+                    updatedParameterKind = updatedParameterKind And Not EnvDTE80.vsCMParameterKind.vsCMParameterKindOut
+
+                Case PARAMETER_PASSING_MODE.cmParameterTypeOut
+                    Throw Exceptions.ThrowEInvalidArg()
```

Comment:
Yeah, I don't like it either. However, striving to throw the same exceptions is something we've done in Code Model to date.

I hate Code Model. :smile: